# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../kbd/hotfix.patch.musl
# Copyright (C) 2019 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- kbd-1.15/src/dumpkeys.c.vanilla	2019-02-08 16:37:59.219179212 +0000
+++ kbd-1.15/src/dumpkeys.c	2019-02-08 16:38:46.371178675 +0000
@@ -3,6 +3,8 @@
  *
  * derived from version 0.81 - aeb@cwi.nl
  */
+
+#define _GNU_SOURCE
 #include <stdio.h>
 #include <stdlib.h>
 #include <ctype.h>
@@ -12,6 +14,7 @@
 #include <linux/kd.h>
 #include <linux/keyboard.h>
 #include <sys/ioctl.h>
+#include <sys/types.h>
 #include <string.h>
 #include <errno.h>
 #include "ksyms.h"
--- kbd-1.15/src/loadkeys.y.vanilla	2019-02-08 16:39:17.487178320 +0000
+++ kbd-1.15/src/loadkeys.y	2019-02-08 16:39:43.407178025 +0000
@@ -18,6 +18,8 @@
 #include <fcntl.h>
 #include <ctype.h>
 #include <sys/ioctl.h>
+#define _GNU_SOURCE
+#include <sys/types.h>
 #include <linux/kd.h>
 #include <linux/keyboard.h>
 #include <unistd.h>		/* readlink */
--- kbd-1.15/src/kbdrate.c.vanilla	2019-02-08 16:40:32.279177468 +0000
+++ kbd-1.15/src/kbdrate.c	2019-02-08 16:42:36.075176056 +0000
@@ -75,6 +75,10 @@
 #include <sys/file.h>
 #include <sys/ioctl.h>
 
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+
 #ifdef __sparc__
 #include <asm/param.h>
 #endif
